android/app/src/androidTest/java/PrnmTest.java

```diff
@@ -0,0 +1,303 @@
+     *
+     * runFirst* sets up a channel, sends TXs and persists it.
+     * runSecond* asserts that the channel can be restored from persistence.
+     */
+
+    Client client;
+    Context ctx = Prnm.contextWithTimeout(120);
+    String prefix;
+    Setup setup;
+
+    AtomicInteger receivedTx = new AtomicInteger(0);
+    AtomicBoolean watcherStopped = new AtomicBoolean(false);
+    // handleConcluded returned
+    AtomicBoolean concluded = new AtomicBoolean(false);
+    AtomicReference<PaymentChannel> ch = new AtomicReference<PaymentChannel>(null);
+
+    @Test
+    /**
+     * Alice entry point for the first test.
+     */
+    public void runFirstTestAlice() throws Exception {
+        runFirstTest(Setup.ALICE);
+    }
+
+    @Test
+    /**
+     * Bob entry point for the test runner to run the first test.
+     */
+    public void runFirstTestBob() throws Exception {
+        // Bob will deploy the contracts on his first start.
+        runFirstTest(Setup.BOB_NO_CONTRACTS);
+    }
+
+    @Test
+    /**
+     * Alice entry point for the test runner to run the second test.
+     */
+    public void runSecondTestAlice() throws Exception {
+        runSecondTest(Setup.ALICE);
+    }
+
+    @Test
+    /**
+     * Bob entry point for the test runner to run the second test.
+     */
+    public void runSecondTestBob() throws Exception {
+        runSecondTest(Setup.BOB);
+    }
+
+    /**
+     * This function does the following:
+     *  both:
+     *      - sharedSetup
+     *      - enables persistence
+     *  Alice:
+     *      - proposes a payment channel to Bob
+     *  Bob:
+     *      - accepts the proposed channel from Alice
+     *  both (repeated 3 times):
+     *      - send 1 TX and wait for a TX from the other peer, Alice goes first
+     *      - exit without closing the channel
+     * @param s
+     * @throws Exception
+     */
+    public void runFirstTest(Setup s) throws Exception {
+        sharedSetup(s);
+        if (s.Index == 0) {
+            proposeChannel(s);
+            sendTx(false, 3);
+        }
+        else {
+            acceptChannel();
+            sendTx(true, 5);
+        }
+        assertBals(106, 94);
+    }
+
+    /**
+     * This function does the following:
+     *  both:
+     *      - sharedSetup
+     *      - restores the channel with persistence
+     *  both (repeated 3 times):
+     *      - send 1 TX and wait for a TX from the other peer, Bob goes first
+     *  Alice:
+     *      - asserts the final balances
+     *  Bob:
+     *      - finalizes the channel
+     *  both:
+     *      - settle the channel
+     * @param s
+     * @throws Exception
+     */
+    public void runSecondTest(Setup s) throws Exception {
+        sharedSetup(s);
+        if (s.Index == 0)
+            sendTx(true, 1);
+        else
+            sendTx(false, 2);
+        sharedTeardown(s);
+
+        assertBals(109, 91);
+        BigInt myBal = client.onChainBalance(ctx, s.Addresses[s.Index]);
```

Comment:
The variable `myBal` is not used anywhere after initialization with the on-chain balance of Alice/Bob.

Reply:
But we maybe could also check the on-chain balances afterward.

Reply:
Yes i forgot that :laughing:

Reply:
There are now `assertOffChainBals` and `assertOnChainBals` functions.